src/resolve.ts

```diff
@@ -90,25 +91,38 @@ function _resolve(id: string | URL, options: ResolveOptions = {}): string {
     : DEFAULT_CONDITIONS_SET;
 
   // Search paths
-  const _urls: URL[] = (
-    (Array.isArray(options.url) ? options.url : [options.url]) as URL[]
-  )
-    .filter(Boolean)
-    .map((url) => new URL(normalizeid(url.toString())));
-  if (_urls.length === 0) {
-    _urls.push(new URL(pathToFileURL(process.cwd())));
-  }
-  const urls = [..._urls];
-  for (const url of _urls) {
-    if (url.protocol === "file:") {
-      urls.push(
-        new URL("./", url),
-        // If url is directory
-        new URL(joinURL(url.pathname, "_index.js"), url),
-        // TODO: Remove in next major version?
-        new URL("node_modules", url),
-      );
+  const urls: URL[] = [];
+  for (const input of Array.isArray(options.url)
+    ? options.url
+    : [options.url]) {
+    if (!input) {
+      continue;
+    }
+    if (input instanceof URL) {
+      urls.push(input);
+      continue;
+    }
+    if (typeof input !== "string") {
+      continue;
+    }
+    if (/^(?:node|data|http|https|file):/.test(input)) {
+      urls.push(new URL(input));
+      continue;
     }
+    try {
+      if (input.endsWith("/") || statSync(input).isDirectory()) {
+        urls.push(url.pathToFileURL(input + "/"));
```

Comment:
For the `input.endsWith("/")` case is it necessary to add another slash?

Reply:
no but it works (node url util normalizes double trailing slash) makes code simpler